ui/design-system-core/src/components/Button/Button.tsx

```diff
@@ -0,0 +1,99 @@
+import React, { PropsWithChildren } from 'react';
+import { tw, apply } from '@twind/core';
+import { Icon } from '../Icon';
+import { IconName } from '../Icon/Icon';
+
+const DefaultIconSize = 'h-4, w-4';
+
+export type ButtonProps = {
+  primary?: boolean;
+  secondary?: boolean;
+  disabled?: boolean;
+  icon?: IconName;
+  iconRight?: boolean;
+  label?: string;
+  loading?: boolean;
+  size?: 'xsmall' | 'small' | 'regular' | 'large';
+  greyBg?: boolean;
+  textonly?: boolean;
+};
+
+const sizing = {
+  xsmall: { textSize: 'text-xs', iconSize: 'h-3 w-3', buttonSize: 'w-24' },
+  small: { textSize: 'text-sm', iconSize: DefaultIconSize, buttonSize: 'w-48' },
+  regular: { textSize: 'text-lg', iconSize: 'h-5, w-5', buttonSize: 'w-64' },
+  large: { textSize: 'text-3xl', iconSize: 'h-7, w-7', buttonSize: 'w-96' },
+};
+
+const baseStyles = apply`text-center flex items-center justify-center	font-medium md:max-md:w-full hover:drop-shadow-md`;
+const textOnlyStyles = apply`
+${baseStyles}
+text-secondary-light dark:text-secondary-dark hover:opacity-50 dark:hover:text-white
+disabled:opacity-50 disabled:cursor-not-allowed space-x-2
+`;
+
+export const Button: React.FC<PropsWithChildren<ButtonProps>> = props => {
+  const { label, size, greyBg, primary } = props;
+  const loading = props.loading === true;
```

Comment:
no need for another variable with the same name for loading, you can use directly the one from props

Reply:
thanks, @quininez for pointing it out. I missed it when doing refactoring 😁